app/controllers/dashboards_controller.rb

```diff
@@ -21,7 +21,7 @@ def show
     fetch_and_authorize_user
     target = @user
     # NOTE: This is a subtle policy check happening here that we are not encapsulating
-    not_authorized if params[:org_id] && !@user.org_admin?(params[:org_id] || @user.any_admin?)
+    not_authorized if params[:org_id] && !(@user.org_admin?(params[:org_id]) || @user.any_admin?)
```

Comment:
when there was an org id param provided,
and neither of these are true
 - user is an org admin of this org
 - user is an admin 
 
then they shouldn't be viewing the organization's dashboard.